### DISCUSSION TITLE
E8 : pkcs11HSM.sh no more available ?

### DISCUSSION THREAD

**Discussion by PhilLaCiotat** [Keyfactor/ejbca-ce#596] [Q&A]
Why the pkcs11HSM.sh script is no more part of the EJBCA8-CE distri ?
It was in EJBCA7.2.11..:
![image](https://github.com/Keyfactor/ejbca-ce/assets/170707436/a11f7e27-5e61-440d-ba6c-3a1fe5ca1c2a)

**Comment by primetomas** [comment]
That script was redundant as it was just calling the same code as "clientToolBox.sh PKCS11HSMKeyTool". You can use that instead.

**Comment by PhilLaCiotat** [comment]
Thanks for your answer. But another effect is that "Test Key" button in GUI from CryptoToken view is no more running if this script is not present. And secondly creating is a key in HSM crypto-token is ko for same reason...
So I suppose I need to add the script again in my release from old version ?

**Reply by primetomas** [reply]
No, that script has nothing to do with functionality in the UI. The UI never calls script for functionality. You have some other issues with your configuration. I recommend running clientToolBox first to verify that the connection to your HSM works.
